library/src/scala/annotation/since.scala

```diff
@@ -0,0 +1,4 @@
+package scala.annotation
+
+/** An annotation that is used to mark symbols added to the stdlib after 3.0 release */
+private[scala] class since(scalaRelease: String) extends scala.annotation.StaticAnnotation
```

Comment:
I presume you have, but have you considered more type-safe alternatives like e.g.?
- `since(majorVersion: Int, minorVersion)`
- `since(scalaRelease: "3.0" | "3.1")`
- `since(scalaRelease: ScalaRelease)`

Reply:
The variant with two ints doesn't really give us much more safety. It would be still possible do use something invalid like `since(1, -100)`.
A union of string literals would require changes to the signature of `since` for every minor release, which is not nice, as `since` is technically a part of stdlib (even though it's private).
`ScalaRelease` can't be used here because it's a part of compiler's internals.
So I think we'll have to live with strings although I've added some additional checks to validate their correctness

Reply:
could this be in the `scala.annotation.internal` package?